### DISCUSSION TITLE
Undercurl does not work

### DISCUSSION THREAD

**Discussion by cd-a** [LazyVim/LazyVim#2369] [Q&A] [answered]
I tried using undercurl with gruvbox, but it doesn't work. it just shows a straight line. (testing on the comments here)

![CleanShot 2024-01-17 at 09 57 33@2x](https://github.com/LazyVim/LazyVim/assets/75008413/fb4bbfa8-9d30-4dff-bca2-fb6e22fd6466)

this is my config. basically all defaults, just the override.
```lua
return {
  -- add gruvbox
  {
    "ellisonleao/gruvbox.nvim",

    opts = {

      terminal_colors = true, -- add neovim terminal colors
      undercurl = true,
      underline = true,
      bold = true,
      italic = {
        strings = true,
        emphasis = true,
        comments = true,
        operators = false,
        folds = true,
      },
      strikethrough = true,
      invert_selection = false,
      invert_signs = false,
      invert_tabline = false,
      invert_intend_guides = false,
      inverse = true, -- invert background for search, diffs, statuslines and errors
      contrast = "", -- can be "hard", "soft" or empty string
      palette_overrides = {},
      overrides = {
        Comment = {
          undercurl = true,
        },
      },
      dim_inactive = false,
      transparent_mode = false,
    },
  },

  -- Configure LazyVim to load gruvbox
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "gruvbox",
    },
  },
}
```

I know that undercurls work in my terminal, because helix shows them like this:
![CleanShot 2024-01-17 at 10 00 57@2x](https://github.com/LazyVim/LazyVim/assets/75008413/934a420f-6239-4a0b-a76f-c0ad4ba8d698)

However, I'm not too certain. I found this on the web to test the terminal, and it shows just a line

![CleanShot 2024-01-17 at 10 03 16@2x](https://github.com/LazyVim/LazyVim/assets/75008413/c4b4ac2b-5685-46c5-9feb-248e22d6de6b)

```bash
printf "\x1b[58;2;255;0;0m\x1b[4msingle\x1b[21mdouble\x1b[60mcurly\x1b[61mdotted\x1b[62mdashed\x1b[0m"
```

however this works

![CleanShot 2024-01-17 at 10 04 11@2x](https://github.com/LazyVim/LazyVim/assets/75008413/25833177-a1fc-4b98-877c-c4647a75a149)

```bash
printf "\e[4:3mhello world\e[0m\n"
```

no idea what's the issue here.

**Comment by dpetka2001** [comment, marked as answer]
This configuration works for me
```lua
  {
    "ellisonleao/gruvbox.nvim",
    opts = {
      -- comment undercurl
      overrides = {
        Comment = {
          undercurl = true,
        },
      },
    },
  },
```
![2024-01-17_15-43](https://github.com/LazyVim/LazyVim/assets/12776461/42e6b6a7-6d6c-455d-9730-dae0c68ce354)

It must be related to your terminal if you're not seeing undercurls. I'm using kitty.

**Reply by cd-a** [reply]
Thanks, I just tested with kitty and it does work indeed. Must be an issue with my Wezterm then. I'll have to investigate

**Reply by dpetka2001** [reply]
@cd-a 

From wezterm [docs](https://wezfurlong.org/wezterm/faq.html#how-do-i-enable-undercurl-curly-underlines) you need to execute the following 
```bash
tempfile=$(mktemp) \
  && curl -o $tempfile https://raw.githubusercontent.com/wez/wezterm/master/termwiz/data/wezterm.terminfo \
  && tic -x -o ~/.terminfo $tempfile \
  && rm $tempfile
```
and then set `term = wezterm` in your `wezterm.lua`. That should take care of it.

**Reply by cd-a** [reply]
that did it, thanks
